File: retail industries Suite -all/lifestyle_fashion_pro/lifestyle_fashion_pro/lifestyle_fashion_management/report/loyalty_points_balance_summary/loyalty_points_balance_summary.py
import frappe


def execute(filters=None):
    columns = [
        {"label": "Customer", "fieldname": "customer", "fieldtype": "Link",
         "options": "Customer", "width": 180},
        {"label": "Current Balance", "fieldname": "balance_points",
         "fieldtype": "Int", "width": 130},
        {"label": "Last Transaction", "fieldname": "transaction_date",
         "fieldtype": "Date", "width": 130},
    ]
    data = frappe.db.sql("""
        SELECT l1.customer, l1.balance_points, l1.transaction_date
        FROM `tabLoyalty Points Ledger` l1
        INNER JOIN (
            SELECT customer, MAX(transaction_date) AS max_date
            FROM `tabLoyalty Points Ledger`
            WHERE docstatus = 1
            GROUP BY customer
        ) l2 ON l1.customer = l2.customer AND l1.transaction_date = l2.max_date
        WHERE l1.docstatus = 1
        ORDER BY l1.balance_points DESC
    """, as_dict=1)
    return columns, data

File: retail industries Suite -all/lifestyle_fashion_pro/lifestyle_fashion_pro/lifestyle_fashion_management/report/loyalty_points_balance_summary/loyalty_points_balance_summary.js
// Copyright (c) 2026, Your Organization and contributors
// For license information, please see license.txt

frappe.query_reports["Loyalty Points Balance Summary"] = {
	filters: [
		{
			fieldname: "customer",
			label: __("Customer"),
			fieldtype: "Link",
			options: "Customer",
			default: "",
		},
		{
			fieldname: "min_balance",
			label: __("Min Balance"),
			fieldtype: "Int",
			default: "",
		},
	],
};
